### PR TITLE
mimix1050rt: ethernet initialization corrected

### LINKED ISSUE #11586: mimxrt1050_evb board: Can't get Ethernet to work
**Describe the bug**
This is continuation of the discussion at (merged/closed) https://github.com/zephyrproject-rtos/zephyr/pull/10875#issuecomment-440625614 .

I cannot get Ethernet connection to BOARD=mimxrt1050_evb. When I connect a patchcord between the mimxrt1050_evb and my laptop, I don't get "link active" LED light up on my laptop side, i.e. it behaves as if the cable wasn't connected. Of course, network interface in Linux doesn't have "RUNNING" status in ifconfig. As discussed at the link above, on mimxrt1050_evb, both Etherjack LEDs are lighted up, and never blink.

This same setup works as expected with frdm_k64f. I.e. if I just switch mimxrt1050_evb with frdm_k64f, leaving the same USB and Ethernet cables, it works, switching back it doesn't, again frdm_k64f - works, back to mimxrt1050_evb - doesn't.

**To Reproduce**

I'm using dumb_http_server as a reference sample to run. 

**Environment (please complete the following information):**
My laptop is Thinkpad X230, with e1000e driver for Ethernet, Ubuntu 18.04:
~~~
Linux x230 4.15.0-39-generic #42-Ubuntu SMP Tue Oct 23 15:48:01 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
~~~

@agansari, Can you please describe you test setup in detail, i.e. what is connected where, etc.

### PR DESCRIPTION
**Pull request by agansari**
Fixes board PHY initialization.
ENET_RST and ENET_INT pins are also used as USER_LED/JTAG_TDI and
INT1_COMBO/JTAG_TDO. Initialization needs to set this pins to power and
interrupt for the PHY Transciver.
Fixes #11586.

Signed-off-by: Andrei Gansari <andrei.gansari@nxp.com>

### CODE REVIEW

**Comment by pfalcon** [comment]
@agansari : Great work! Will test later today.

Minor issue in the commit message:

> Initialization needs to set this pins

Should be "these pins".

And well, need to do something about that completely unrelated test escaping 128K of code space. I didn't receive even comments on https://github.com/zephyrproject-rtos/zephyr/issues/11548, but apparently, that needs to be done sooner rather than later, please see if you can get to it sooner than me, as I didn't even get to needed stuff in me very intermittent reading of i.MX1052 TRM.

File: boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst [hunk 1]
```diff
@@ -129,7 +129,9 @@ The MIMXRT1050 SoC has five pairs of pinmux/gpio controllers.
 +---------------+-----------------+---------------------------+
```
Comment (jukkar) [inline boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst]: Typos in commit msg

s/etherent/ethernet/
s/corected/corrected/

**Review by jukkar — CHANGES_REQUESTED** [review]

**Comment by pfalcon** [comment]
This patch actually makes it worse - the link guaranteedly stays down for me. I describe my testing in a series of comments in https://github.com/zephyrproject-rtos/zephyr/issues/11586#issuecomment-445432354 . 

I thus recommend not to merge it, because with it, the ethernet 100% doesn't work for some people, whereas without this patch, at least link can be established more or less deterministically and board->host TX packets work.

**Review by dbkinder — APPROVED** [review]
+1 for doc changes

**Comment by pfalcon** [comment]
Now, just enabling CONFIG_NET_SHELL=y in dumb_http_server leads to dead link again.

**Comment by pfalcon** [comment]
Size of dumb_http_server with CONFIG_NET_SHELL=n: 69712, CONFIG_NET_SHELL=y: 98012. So, I wouldn't blame code size on the issue, unless imxrt memory setup is really wrong.

**Comment by pfalcon** [comment]
But now what I do is set `CONFIG_SHELL_BACKEND_SERIAL_INTERRUPT_DRIVEN=n`. In other words, I have:

~~~
+CONFIG_NET_SHELL=y
+CONFIG_SHELL_BACKEND_SERIAL_INTERRUPT_DRIVEN=n
~~~

And, bingo - Eth link is up.

@nordic-krch: I told you adding that option won't be wasted effort! cc: @jarz-nordic too.

@agansari: So, it looks like there's some conflict with UART [interrupts] and ETH [interrupts]. Can you please look into that?

**Comment by pfalcon** [comment]
Enabling also CONFIG_NET_STATISTICS=y, I confirm that the board doesn't receive any packets. Trying to do ping from board, the host sees ARP request for 192.0.2.2, but the board doesn't see reply to it.

@agansari, So, we made a full iteration and back to square: RX doesn't work for me, like it didn't before. But at least Eth link establishment now works reliably, and we know how to reproduce a failure when it doesn't.

**Comment by pfalcon** [comment]
And now knowing the cause of link-down cases, as far as I can tell, "k_busy_wait(USEC_PER_MSEC);" doesn't make any difference, so I'd suggest dropping this second commit here, let's vote this PR up as it goes in the right direction, and let's continue with investigation why RX interrupts still don't fire for some folks.

**Review by jukkar — APPROVED** [review]

**Review by pfalcon — CHANGES_REQUESTED** [review]
Let's remove busy-waiting (unless there're indications that it's really needed). Giving -1 so it wasn't merged by accident with that.

**Review by pfalcon — APPROVED** [review]

**Comment by pfalcon** [comment]
recheck

**Comment by pfalcon** [comment]
This fails CI due to unrelated issue:
~~~
-- Loading /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/boards/arm/nucleo_f756zg/nucleo_f756zg.dts as base
-- Overlaying /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/dts/common/common.dts
nucleo_f756zg.dts_compiled: ERROR (duplicate_label): /flash@8000000: Duplicate label 'flash0' on /flash@8000000 and /soc/flash-controller@40023c00/flash@8000000
ERROR: Input tree has errors, aborting (use -f to force output)
CMake Error at /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/cmake/dts.cmake:130 (message):
  command failed with return code: 2
~~~

@agansari, Can you please rebase to see it that helps?

**Comment by agansari** [comment]
@pfalcon rebase worked

**Pull request merged**